src/plugins/data/public/query/query_string/language_service/lib/query_result.tsx

```diff
@@ -58,6 +68,43 @@ export function QueryResult(props: { queryStatus: QueryStatus }) {
     };
   }, [props.queryStatus.startTime]);
 
+  const displayErrorMessage = useMemo(() => {
+    const error = props.queryStatus.body?.error;
+    const message = error?.message;
+
+    if (message == null) {
+      if (typeof error === 'string') {
+        return error;
+      }
+
+      if (typeof error === 'object') {
+        return JSON.stringify(error);
+      }
+
+      return `Unknown Error: ${String(error)}`;
+    }
+
+    // For async search strategy, expecting message.error to be string
+    if (typeof message.error === 'string') {
+      return message.error;
+    }
+
+    // For normal search strategy, expecting message.error to be object
+    if (message.error?.details) {
+      return message.error.details;
+    }
+
+    if (typeof message === 'string') {
+      return message;
+    }
+
+    if (typeof message === 'object') {
+      return JSON.stringify(message);
+    }
+
+    return `Unknown Error: ${String(message)}`;
+  }, [props.queryStatus.body?.error]);
```

Comment:
I feel that this can be extracted into a util function, which will also make it easier to unit test. If you want to keep the memoization:

```
const displayErrorMessage = useMemo(() => getFormattedErrorMessageOrSomething(props.queryStatus.body?.error), [props.queryStatus]);
```